### PR TITLE
MP-9129 - Improve the Soft Delete logic of the Auditable Entities with compound Nox Types

### LINKED ISSUE #725: [Nox.Lib] Improve the Soft Delete logic of the Auditable Entities with compound Nox Types [MP-9129]

### PR DESCRIPTION
**Pull request by dejobratic**
- Updated DbContext to reattach the deleted compound types when the parent entity is marked for deletion.
- Changed the store.entity.nox.yaml file:
    - Changed the id from nuid to databaseGuid, because the issue was observed on entities with databaseGuid as keys
    - Added multiple compound types, to make sure the reattaching logic works as expected
- Added tests for confirming that the soft delete works as expected

### CODE REVIEW

File: src/Nox.Generator/Infrastructure/Persistence/DbContext.template.cs [hunk 1]
```diff
@@ -134,10 +135,25 @@ private void AuditEntity(EntityEntry<AuditableEntityBase> entry)
...
+                ReattachCompoundTypes(entry);
                 break;
         }
     }
 
+    private void ReattachCompoundTypes(EntityEntry<AuditableEntityBase> parentEntry)
+    {
+        foreach (var navigationEntry in parentEntry.Navigations)
+        {
+            foreach (var typeEntry in ChangeTracker.Entries<INoxType>())
```
Comment (rochar) [inline src/Nox.Generator/Infrastructure/Persistence/DbContext.template.cs]: @dejobratic  why INoxType and not IOwnedEntity? INoxType is for fields

File: src/Nox.ClientApi.Tests/.nox/design/Entitities/store.entity.nox.yaml [hunk 2]
```diff
@@ -11,13 +11,8 @@ description: Stores
 keys:
 
   - name: Id
+    type: databaseGuid
     isRequired: true
-    description: NuidField Type
```
Comment (rochar) [inline src/Nox.ClientApi.Tests/.nox/design/Entitities/store.entity.nox.yaml:15]: why the change, this way we lose the tests regarding Nuid?

File: src/Nox.ClientApi.Tests/Tests/Controllers/StoresControllerTests.cs
Comment (rochar) [inline src/Nox.ClientApi.Tests/Tests/Controllers/StoresControllerTests.cs:1]: Can you also add test on Countries that does soft delete and thens tries to get a owned entity via /api/Countries/{key}/CountryLocalName/{relatedKey}, should return not found => ensure owned of soft deleted entities are not retrievable

**Review by rochar — CHANGES_REQUESTED** [review]

File: src/Nox.Generator/Infrastructure/Persistence/DbContext.template.cs (hunk 1 above)
Comment (dejobratic) [reply src/Nox.Generator/Infrastructure/Persistence/DbContext.template.cs]: The issue has been observed for Nox compound types, like StreetAddress and TranslatedText, which are registered in EF as owned entities, but they are of INoxType.

File: src/Nox.ClientApi.Tests/.nox/design/Entitities/store.entity.nox.yaml (hunk 2 above)
Comment (dejobratic) [reply src/Nox.ClientApi.Tests/.nox/design/Entitities/store.entity.nox.yaml:15]: We already have workplace.entity.nox.yaml with the same nuid key and configuration. 
image

And if I saw correctly, it has Nuid tests.
image

Therefore I decided to change the Store key to databaseGuid.

File: src/Nox.Generator/Infrastructure/Persistence/DbContext.template.cs (hunk 1 above)
Comment (rochar) [reply src/Nox.Generator/Infrastructure/Persistence/DbContext.template.cs]: and other owned entities, for example CountryLocalNames in Countries, do we have the same issue?

File: src/Nox.ClientApi.Tests/.nox/design/Entitities/store.entity.nox.yaml (hunk 2 above)
Comment (rochar) [reply src/Nox.ClientApi.Tests/.nox/design/Entitities/store.entity.nox.yaml:15]: ok,

File: src/Nox.ClientApi.Tests/Tests/Controllers/StoresControllerTests.cs
Comment (dejobratic) [reply src/Nox.ClientApi.Tests/Tests/Controllers/StoresControllerTests.cs:1]: Test case has been added. Thanks for the suggestion.

File: src/Nox.Generator/Infrastructure/Persistence/DbContext.template.cs (hunk 1 above)
Comment (dejobratic) [reply src/Nox.Generator/Infrastructure/Persistence/DbContext.template.cs]: You are right. Those cases have been covered now.

**Review by rochar — APPROVED** [review]

**Review by gjorgji-grgovski-authoritypartners — APPROVED** [review]
